langchain/src/agents/structured_chat/index.ts

```diff
@@ -0,0 +1,143 @@
+import { zodToJsonSchema } from "zod-to-json-schema";
+import { JsonSchema7ObjectType } from "zod-to-json-schema/src/parsers/object.js";
+
+import { BaseLanguageModel } from "../../base_language/index.js";
+import { LLMChain } from "../../chains/llm_chain.js";
+import { PromptTemplate } from "../../prompts/prompt.js";
+import {
+  ChatPromptTemplate,
+  HumanMessagePromptTemplate,
+  SystemMessagePromptTemplate,
+} from "../../prompts/chat.js";
+import { AgentStep } from "../../schema/index.js";
+import { StructuredTool } from "../../tools/base.js";
+import { Optional } from "../../types/type-utils.js";
+import { Agent, AgentArgs, OutputParserArgs } from "../agent.js";
+import { AgentInput } from "../types.js";
+import { StructuredChatOutputParserWithRetries } from "./outputParser.js";
+import { FORMAT_INSTRUCTIONS, PREFIX, SUFFIX } from "./prompt.js";
+
+export interface ChatCreatePromptArgs {
+  /** String to put after the list of tools. */
+  suffix?: string;
+  /** String to put before the list of tools. */
+  prefix?: string;
+  /** List of input variables the final prompt will expect. */
+  inputVariables?: string[];
+}
+
+export type ChatAgentInput = Optional<AgentInput, "outputParser">;
+
+/**
+ * Agent that interoperates with Structured Tools using React logic.
+ * @augments Agent
+ */
+export class StructuredChatAgent extends Agent {
```

Comment:
Can we add this new agent to the list of agents tested in `evaluation.int.test` ?